lib/bitcoin/BitcoinFileReader.ts

```diff
@@ -0,0 +1,35 @@
+import IBitcoinFileReader from './interfaces/IBitcoinFileReader';
+import * as fs from 'fs';
+
+/**
+ * concrete implementation of BitcoinFileReader
+ */
+export default class BitcoinFileReader implements IBitcoinFileReader {
+  /**
+   * Constructor
+   * @param bitcoinDataDirectory The same directory used by bitcoin core
+   */
+  public constructor (public bitcoinDataDirectory: string) {}
+
+  public listBlockFiles (): string[] {
+    const blocksDataDirectoryPath = `${this.bitcoinDataDirectory}/blocks`;
+    let blockDataDir;
+    try {
+      blockDataDir = fs.readdirSync(blocksDataDirectoryPath);
+    } catch (e) {
+      console.error(`Error thrown while reading file system: ${e}`);
+      return [];
```

Comment:
Sanity: wouldn't return [] potentially allow initialization code silently thinking all files are processed? Of course unless it checks for [] explicitly and fail, but then we can do that here?

Reply:
Will throw instead of return default.